config/routes.js

```diff
@@ -1,93 +1,105 @@
+  }), users.generateToken, users.returnToken);
+
+  app.get('/api/users/me', users.me);
+
+  app.get('/users/me', customAuth.hasAuth(), users.me);
+  app.get('/users/:userId', users.show);
+
+    // Setting the facebook oauth routes
+  app.get('/auth/facebook', passport.authenticate('facebook', {
+    scope: ['email'],
+    failureRedirect: '/signin'
+  }), users.signin);
+
+  app.get('/auth/facebook/callback', passport.authenticate('facebook', {
+    failureRedirect: '/signin'
+  }), users.authCallback);
+
+    // Setting the github oauth routes
+  app.get('/auth/github', passport.authenticate('github', {
+    failureRedirect: '/signin'
+  }), users.signin);
+
+  app.get('/auth/github/callback', passport.authenticate('github', {
+    failureRedirect: '/signin'
+  }), users.authCallback);
+
+    // Setting the twitter oauth routes
+  app.get('/auth/twitter', passport.authenticate('twitter', {
+    failureRedirect: '/signin'
+  }), users.signin);
+
+  app.get('/auth/twitter/callback', passport.authenticate('twitter', {
+    failureRedirect: '/signin'
+  }), users.authCallback);
+
+    // Setting the google oauth routes
+  app.get('/auth/google', passport.authenticate('google', {
+    failureRedirect: '/signin',
+    scope: [
+      'https://www.googleapis.com/auth/userinfo.profile',
+      'https://www.googleapis.com/auth/userinfo.email'
+    ]
+  }), users.signin);
+
+  app.get('/auth/google/callback', passport.authenticate('google', {
+    failureRedirect: '/signin'
+  }), users.authCallback);
+
+    // Finish with setting up the userId param
+  app.param('userId', users.user);
 
     // Answer Routes
-    var answers = require('../app/controllers/answers');
-    app.get('/answers', answers.all);
-    app.get('/answers/:answerId', answers.show);
+  const answers = require('../app/controllers/answers');
+  app.get('/answers', answers.all);
+  app.get('/answers/:answerId', answers.show);
     // Finish with setting up the answerId param
-    app.param('answerId', answers.answer);
+  app.param('answerId', answers.answer);
 
     // Question Routes
-    var questions = require('../app/controllers/questions');
-    app.get('/questions', questions.all);
-    app.get('/questions/:questionId', questions.show);
+  const questions = require('../app/controllers/questions');
+  app.get('/questions', questions.all);
+  app.get('/questions/:questionId', questions.show);
     // Finish with setting up the questionId param
-    app.param('questionId', questions.question);
+  app.param('questionId', questions.question);
 
     // Avatar Routes
-    var avatars = require('../app/controllers/avatars');
-    app.get('/avatars', avatars.allJSON);
+  const avatars = require('../app/controllers/avatars');
+  app.get('/avatars', avatars.allJSON);
 
-    //Home route
-    var index = require('../app/controllers/index');
-    app.get('/play', index.play);
-    app.get('/', index.render);
+    // Home route
+  const index = require('../app/controllers/index');
```

Comment:
Any specific reason why the packages are required within the code as opposed to it all being done at the beginning of the file?! If so, please leave a comment as to why.